packages/hash/frontend/src/components/hooks/blockProtocolFunctions/ontology/useBlockProtocolAggregateDataTypes.ts

```diff
@@ -0,0 +1,60 @@
+import { useLazyQuery } from "@apollo/client";
+
+import { useCallback } from "react";
+import {
+  GetAllLatestDataTypesQuery,
+  GetAllLatestDataTypesQueryVariables,
+} from "../../../../graphql/apiTypes.gen";
+import { getAllLatestDataTypesQuery } from "../../../../graphql/queries/ontology/data-type.queries";
+import { AggregateDataTypesMessageCallback } from "./ontology-types-shim";
+
+export const useBlockProtocolAggregateDataTypes = (): {
+  aggregateDataTypes: AggregateDataTypesMessageCallback;
+} => {
+  const [aggregateFn] = useLazyQuery<
+    GetAllLatestDataTypesQuery,
+    GetAllLatestDataTypesQueryVariables
+  >(getAllLatestDataTypesQuery, {
+    fetchPolicy: "no-cache",
+  });
+
+  const aggregateDataTypes = useCallback<AggregateDataTypesMessageCallback>(
+    async ({ data }) => {
+      if (!data) {
+        return {
+          errors: [
+            {
+              code: "INVALID_INPUT",
+              message: "'data' must be provided for aggregateDataTypes",
+            },
+          ],
+        };
+      }
+
+      /** @todo Add filtering to this aggregate query. */
+      const response = await aggregateFn({
+        query: getAllLatestDataTypesQuery,
+      });
+
+      if (!response.data) {
+        return {
+          errors: [
+            {
+              code: "INVALID_INPUT",
+              message: "Error calling aggregateDataTypes",
+            },
+          ],
+        };
+      }
+
+      return {
+        data: {
+          results: response.data.getAllLatestDataTypes,
+        },
+      };
+    },
+    [aggregateFn],
+  );
+
+  return { aggregateDataTypes };
```

Comment:
Curious why we don't just return the function itself?

Reply:
This was more or less convention from the other functions. I assume it helps enforce that we don't rename the methods when using the "hook"

Reply:
The convention with hooks is to let the consumer name the return value. There are *some* exceptions. I don't think we need to change this now but I do think we should look into it across the board.